pkg/hs/urbit-king/lib/Urbit/Vere/Eyre/KingSubsite.hs

```diff
@@ -0,0 +1,113 @@
+{-|
+  KingSubsite: runtime-exclusive HTTP request handling, for /~_~
+-}
+
+module Urbit.Vere.Eyre.KingSubsite
+  ( KingSubsite
+  , kingSubsite
+  , runKingSubsite
+  , fourOhFourSubsite
+  ) where
+
+import Urbit.Prelude hiding (Builder)
+
+import Data.ByteString.Builder
+import Urbit.King.Scry
+import Urbit.Vere.Serf.Types
+
+import Data.Conduit       (ConduitT, Flush(..), yield)
+import Data.Text.Encoding (encodeUtf8Builder)
+
+import qualified Data.Text.Encoding  as E
+import qualified Network.HTTP.Types  as H
+import qualified Network.Wai         as W
+import qualified Network.Wai.Conduit as W
+import qualified Urbit.Noun.Time     as Time
+
+newtype KingSubsite = KS { runKingSubsite :: W.Application }
+
+data SlogAction
+  = KeepAlive
+  | Slog (Atom, Tank)
+
+streamSlog :: Monad m => SlogAction -> ConduitT () (Flush Builder) m ()
+streamSlog a = do
+  case a of
+    KeepAlive -> pure ()
+    Slog (_, t) -> for_ (wash (WashCfg 0 80) (tankTree t)) $ \l -> do
+      yield $ Chunk "data:"
+      yield $ Chunk $ encodeUtf8Builder $ unTape l
+      yield $ Chunk "\n"
+  yield $ Chunk "\n"
+  yield $ Flush
+
+kingSubsite :: HasLogFunc e
+            => Ship
+            -> (Time.Wen -> Gang -> Path -> IO (Maybe (Term, Noun)))
+            -> TVar ((Atom, Tank) -> IO ())
+            -> RAcquire e KingSubsite
+kingSubsite who scry func = do  --TODO  unify who into scry
```

Comment:
what's this todo?

Reply:
I thought we might factor this out by passing it into `scry` early, but idk if that's really worth it. Removing.